test/resources/complex_xml_transformed.yml

```diff
@@ -0,0 +1,626 @@
+      stadatfinpresta: '2023-10-26'
+      tcategory_integratedstage: lqvDg6O5jw7QDFE
+      ncategory_integratedstage: kDI6Bnk5qbUN64B
+      mcategory_integratedstage: e0fxG2RfYosEFE5
+      tnmstagegrouping_integrated: oOBlEwwH3GtZKWQ
+      org_site_tnm_group_int: 3KLUG
+      stadatintsta: '2023-10-26'
+      tnm_coding_edition: '1'
+      tnmedition: QL
+      :rawtext:
+        tcategory_finalpretreatment: ngv1KRgf5WOXuc7
+        ncategory_finalpretreatment: V8h1It3jLl4Izvf
+        mcategory_finalpretreatment: 483wBKjug9MRB0l
+        tnmstagegrouping_finalpretreatment: o5aUmT28ffUyKWj
+        org_site_tnm_group_final_pt: LN0ZH
+        tnmstagegroupingdate_finalpretreatment: '2023-10-26'
+        tcategory_integratedstage: lqvDg6O5jw7QDFE
+        ncategory_integratedstage: kDI6Bnk5qbUN64B
+        mcategory_integratedstage: e0fxG2RfYosEFE5
+        tnmstagegrouping_integrated: oOBlEwwH3GtZKWQ
+        org_site_tnm_group_int: 3KLUG
+        tnmstagegroupingdate_integrated: '2023-10-26'
+        tnm_coding_edition: '1'
+        tnmeditionnumber: QL
+        changstagingformedulloblastoma: ''
+        wilmstumourstage: ''
+        intl_risk_group_staging_system_stage: ''
+        pretextstagingsystemstage: ''
+        pretextannotationfactors: ''
+        internationalstagingsystemforretinoblastoma: ''
+        annarborstage: ''
+        binetstage: ''
+        rissstageformyeloma: ''
+        murphy_stjudestage: ''
+        childrens_oncology_group_cog_staging_system_stage: ''
+        central_nervous_system_involvement: ''
+        barcelonacliniclivercancerstagebclcstage: ''
+        stagegrouping_testicular: ''
+        annarborsymptoms: ''
+        annarborextranodality: ''
+        annarborbulk: ''
+        annarborsplenicinvolvement: ''
+    - 0
+  - - ECosdXmlRecurrence#NPP
+    - :rawtext:
+        dateofnonprimarydiagnosis: ''
+        originalprimarydiagnosis_icd: ''
+        metastatictype: ''
+        metastaticsite: ''
+    - 0
+  - - ECosdXmlProgression#NPP
+    - :rawtext:
+        dateofnonprimarydiagnosis: ''
+        cancer_progression_icd: ''
+        metastatictype: ''
+        metastaticsite: ''
+    - 0
+  - - ECosdXmlRecurrence
+    - :rawtext:
+        palliativecarespecialistseenindicator_cancerrecurrence: ''
+        relapse_method_of_detection: ''
+    - 0
+  - - ECosdXmlProgression#1
+    - metastatictype_1: '02'
+      metastaticsite_1: '07'
+      dateofnonprimarydiagnosis: '2023-10-26'
+      :rawtext:
+        metastatictype_1: '02'
+        metastaticsite_1: '07'
+        dateofnonprimarydiagnosis: '2023-10-26'
+    - 0
+  - - ECosdXmlProgression#2
+    - metastatictype_3: '01'
+      metastaticsite_3: '09'
+      metastatictype_4: '03'
+      metastaticsite_4: '02'
+      dateofnonprimarydiagnosis: '2023-10-26'
+      :rawtext:
+        metastatictype_3: '01'
+        metastaticsite_3: '09'
+        metastatictype_4: '03'
+        metastaticsite_4: '02'
+        dateofnonprimarydiagnosis: '2023-10-26'

```

Comment:
Is it expected behaviour that these are `metastatictype_3` and `metastatictype_4` (with no `metastatictype_1` or `metastatictype_2` in this block)?

The entry above seems to have `metastatictype_1`, and there's a `metastatictype_2` somewhere else.

Reply:
Yep, agreed this does look odd. 

To ensure all data is mapped, we needed unique increments per klass. I get those by calculating the sum of the numbers in the xpaths.

Example xpaths for the above secnario you obeserved:

```
- PrimaryPathway/Diagnosis/Progression[1]/MetastaticTypeAndSite/MetastaticType/@code
- PrimaryPathway/Diagnosis/Progression[1]/MetastaticTypeAndSite/MetastaticSite/@code
- PrimaryPathway/Diagnosis/Progression[2]/MetastaticTypeAndSite[1]/MetastaticType/@code
- PrimaryPathway/Diagnosis/Progression[2]/MetastaticTypeAndSite[1]/MetastaticSite/@code
- PrimaryPathway/Diagnosis/Progression[2]/MetastaticTypeAndSite[2]/MetastaticType/@code
- PrimaryPathway/Diagnosis/Progression[2]/MetastaticTypeAndSite[2]/MetastaticSite/@code
```

The first instance of progression section, `PrimaryPathway/Diagnosis/Progression[1]`, only has one pair of `MetastaticType` and `MetastaticSite`, so the sum of the numbers in the xpath is 1.

The second progression section `PrimaryPathway/Diagnosis/Progression[2]` has two pairs, the first pair has `[2]` and `[1]` in the xpath so the increment is 3. The second pair has `[2]` and `[2], so the increment is 4.


The following xpaths...


```
- PrimaryPathway/Diagnosis/Progression[1]/MetastaticTypeAndSite[1]/MetastaticType/@code
- PrimaryPathway/Diagnosis/Progression[1]/MetastaticTypeAndSite[1]/MetastaticSite/@code
- PrimaryPathway/Diagnosis/Progression[1]/MetastaticTypeAndSite[2]/MetastaticType/@code
- PrimaryPathway/Diagnosis/Progression[1]/MetastaticTypeAndSite[2]/MetastaticSite/@code
- PrimaryPathway/Diagnosis/Progression[2]/MetastaticTypeAndSite[1]/MetastaticType/@code
- PrimaryPathway/Diagnosis/Progression[2]/MetastaticTypeAndSite[1]/MetastaticSite/@code
- PrimaryPathway/Diagnosis/Progression[2]/MetastaticTypeAndSite[2]/MetastaticType/@code
- PrimaryPathway/Diagnosis/Progression[2]/MetastaticTypeAndSite[2]/MetastaticSite/@code
```
...would produce something like:

```
 - - ECosdXmlProgression#1
    - metastatictype_2: '01'
      metastaticsite_2: '09'
      metastatictype_3: '03'
      metastaticsite_3: '02'
      dateofnonprimarydiagnosis: '2023-10-26'
      :rawtext:
        metastatictype_2: '01'
        metastaticsite_2: '09'
        metastatictype_3: '03'
        metastaticsite_3: '02'
 - - ECosdXmlProgression#2
    - metastatictype_3: '01'
      metastaticsite_3: '09'
      metastatictype_4: '03'
      metastaticsite_4: '02'
      dateofnonprimarydiagnosis: '2023-10-26'
      :rawtext:
        metastatictype_3: '01'
        metastaticsite_3: '09'
        metastatictype_4: '03'
        metastaticsite_4: '02'
```

Reply:
As long as we know this is an intermediate representation, and the downstream code can deal with missing offsets, then I'm happy with this.

Reply:
Yes, absolutely - this is an intermediate representation that would rely on downstream logic to transform it further. The main objecive of these changes was to a) avoid verbosely creating table mappings for all possible xpaths, and b) to ensure data remains "grouped".